### ISSUE TITLE
converting categorical 

### ISSUE THREAD

**Issue by sharikalog7** [sharikalog7/Machine-learning#1]
# In[1]:

import pandas as pd
import numpy as np 
import tensorflow as tf

from tensorflow import feature_column
from tensorflow.keras import layers
from sklearn.model_selection import train_test_split

# In[2]:

#create dataframe

df=pd.read_csv("diabetic_data.csv")

# In[3]:

##shape of categorical,int,float

df.select_dtypes(include=['object']).shape,df.select_dtypes(include=['int64']).shape,df.select_dtypes(include=['float']).shape

# In[4]:

df.head()

# In[5]:

####dropping categorical columns
df=df.drop(columns=['race','gender','age','weight','payer_code','diag_1','diag_2','diag_3','max_glu_serum','A1Cresult','metformin','repaglinide','nateglinide','chlorpropamide','glimepiride','acetohexamide','glipizide','glyburide','tolbutamide','pioglitazone','rosiglitazone','acarbose','miglitol','troglitazone','tolazamide','insulin','glyburide-metformin','glipizide-metformin','glimepiride-pioglitazone','metformin-rosiglitazone','metformin-pioglitazone','medical_specialty','examide','citoglipton','readmitted'])

# In[6]:

##shape after dropping categorical column .
df.select_dtypes(include=['object']).shape,df.select_dtypes(include=['int64']).shape,df.select_dtypes(include=['float']).shape

# In[7]:

#splitting into train,test,validation
train, test = train_test_split(df, test_size=0.2)
train, val = train_test_split(train, test_size=0.2)
print(len(train), 'train examples')
print(len(val), 'validation examples')
print(len(test), 'test examples')

# In[8]:

###converting target using Label Encoder 

from sklearn.preprocessing import LabelEncoder

lb_make = LabelEncoder()
df["diabetesMed"] = lb_make.fit_transform(df["diabetesMed"])
df[["diabetesMed", "diabetesMed"]].head(11)

# In[9]:

# A utility method to create a tf.data dataset from a Pandas Dataframe
def df_to_dataset(df, shuffle=True, batch_size=32):
  df = df.copy()
  labels = df.pop('diabetesMed')
  ds = tf.data.Dataset.from_tensor_slices((dict(df), labels))
  if shuffle:
    ds = ds.shuffle(buffer_size=len(df))
  ds = ds.batch(batch_size)
  return ds

# In[10]:

batch_size = 50 # A small batch sized is used for demonstration purposes
train_ds = df_to_dataset(train, batch_size=batch_size)
val_ds = df_to_dataset(val, shuffle=False, batch_size=batch_size)
test_ds = df_to_dataset(test, shuffle=False, batch_size=batch_size)

# In[11]:

for feature_batch, label_batch in train_ds.take(1):
  print('Every feature:', list(feature_batch.keys()))
  print('A batch of ages:', feature_batch)
  print('A batch of targets:', label_batch)

# In[12]:

# We will use this batch to demonstrate several types of feature columns
example_batch = next(iter(train_ds))[0]

# In[13]:

# A utility method to create a feature column
# and to transform a batch of data

def demo(feature_column):
  feature_layer = layers.DenseFeatures(feature_column)
  print(feature_layer(example_batch).numpy())

# In[14]:

###converting categorical columns which have only upto 3 values 
#for checking converted only one feature categorical column
feature_columns = []

# indicator cols
change = feature_column.categorical_column_with_vocabulary_list(
      'change', ['No', 'Ch'])
change_code = feature_column.indicator_column(change)
feature_columns.append(change_code)

# In[15]:

feature_layer = tf.keras.layers.DenseFeatures(feature_columns)

# In[16]:

batch_size = 32
train_ds = df_to_dataset(train, batch_size=batch_size)
val_ds = df_to_dataset(val, shuffle=False, batch_size=batch_size)
test_ds = df_to_dataset(test, shuffle=False, batch_size=batch_size)

# In[17]:

model = tf.keras.Sequential([
  feature_layer,
  layers.Dense(128, activation='relu'),
  layers.Dense(128, activation='relu'),
  layers.Dense(1)
])

model.compile(optimizer='adam',
              loss=tf.keras.losses.BinaryCrossentropy(from_logits=True),
              metrics=['accuracy'])

model.fit(train_ds,
          validation_data=val_ds,
          epochs=10)

# In[ ]:

model = get_compiled_model()
model.fit(train_ds, epochs=15)

# In[ ]:

loss, accuracy = model.evaluate(test_ds)
print("Accuracy", accuracy)
